src/NServiceBus.Core/MessageInterfaces/MessageMapper/Reflection/MessageMapper.cs

```diff
@@ -44,28 +44,48 @@ public void Initialize(IEnumerable<Type> types)
         public Type GetMappedTypeFor(Type t)
         {
             Guard.AgainstNull(nameof(t), t);
+
+            Type mappedType;
+            if (TryGetMappedType(t, out mappedType))
+            {
+                return mappedType;
+            }
+
+            InitType(t);
+
+            return TryGetMappedType(t, out mappedType) ? mappedType : null;
```

Comment:
Why not just call `TryGetMapped` and then return mappedType, since that will already be `null` when false?